src/cpp/session/SessionModuleContext.cpp

```diff
@@ -84,6 +89,63 @@ namespace module_context {
       
 namespace {
 
+// simple service for handling console_input rpc requests
+class ConsoleInputService : boost::noncopyable
```

Comment:
This looks more or less identical to what we're doing for notebooks -- do you think it'd be worth factoring such that code can be shared between the implementations?

https://github.com/rstudio/rstudio/blob/b6e2de8d6248f522d9df4b9fcd2a74557f76171f/src/cpp/session/modules/rmarkdown/NotebookQueue.cpp#L468-L490

Reply:
Good idea!

Reply:
I looked into refactoring this but I don't understand the Notebook queue internals well enough to make that change with confidence -- I'm going to leave this for now.